components/tableIndexer/table_constructor.py

```diff
@@ -0,0 +1,154 @@
+from transformers import TableTransformerForObjectDetection
+import torch
+from torchvision import transforms
+from PIL import Image, ImageDraw
+from utils.img_processing import MaxResize, outputs_to_objects
+from transformers import TableTransformerForObjectDetection
+from utils.logger import setup_logging
+import logging
+import argparse
+import os
+import json as js
+import sys
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Construct tables by defining rows for detected tables."
+    )
+    parser.add_argument(
+        "--log-level",
+        dest="log_level",
+        default="INFO",
+        help="Set the logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)",
+    )
+    parser.add_argument(
+        "--img-path",
+        dest="img_path",
+        required=True,
+        help="Specify the path of the cropped table",
+    )
+    parser.add_argument(
+        "--model-name",
+        dest="model_name",
+        default="microsoft/table-structure-recognition-v1.1-all",
+        help="Specify the path of the cropped table",
+    )
+    parser.add_argument(
+        "--device",
+        dest="device",
+        default="cpu",
+        help="Specify the device type to load the model (cpu, cuda, mkldnn, opengl, opencl, ideep, hip, msnpu",
+    )
+    parser.add_argument(
+        "--output-directory",
+        dest="output_directory",
+        default="./outputs/constructed_tables",
+        help="Specify the output directory for images",
+    )
+    parser.add_argument(
+        "--json", dest="json", default=False, help="Save output as a json file"
+    )
+    parser.add_argument(
+        "--json-path",
+        dest="json_path",
+        default="./outputs/constructed_tables_json",
+        help="output directory for the json file",
+    )
+    return parser.parse_args()
+
+
+def table_constructor(
```

Comment:
```suggestion
def table_constructor(*, img_path,....
```
Because the arguments are mostly strings it might be easy to mix up which arguments are which unless you used named arugments